contract/src/lib.rs

```diff
@@ -423,4 +409,28 @@ impl MpcContract {
         self.pending_requests.remove(payload);
         self.request_counter -= 1;
     }
+
+    // Helper functions
+    #[private]
+    #[init(ignore_state)]
+    pub fn clean(keys: Vec<near_sdk::json_types::Base64VecU8>) -> Self {
+        log!("clean: keys={:?}", keys);
+        for key in keys.iter() {
+            env::storage_remove(&key.0);
+        }
+        Self {
+            protocol_state: ProtocolContractState::NotInitialized,
+            pending_requests: LookupMap::new(b"m"),
+            request_counter: 0,
+        }
+    }
+
+    #[private]
+    pub fn clean_payloads(&mut self, payloads: Vec<[u8; 32]>, counter: u32) {
```

Comment:
why specify the payloads instead of clearing all the requests in the pending_requests?

Reply:
I've used ```
#[private]
    pub fn clean_payloads(&mut self) {
        log!("clean_payloads");
        self.pending_requests = LookupMap::new(b"m");
        self.request_counter = 0;
    }
```
to clean the payloads on dev for now.

Reply:
Ok, that is fine. But the old payloads are still there.
LookupMap can not clean its key-value pairs. It is optimized for storage and access speed.

Reply:
wait, why would the old payloads still be there? I thought `self.pending_requests = LookupMap::new(b"m");
        self.request_counter = 0;` will reset it?

Reply:
This is how insert and remove works in `LookupMap`
```rust
    /// Inserts a serialized key-value pair into the map.
    /// If the map did not have this key present, `None` is returned. Otherwise returns
    /// a serialized value. Note, the keys that have the same hash value are undistinguished by
    /// the implementation.
    pub fn insert_raw(&mut self, key_raw: &[u8], value_raw: &[u8]) -> Option<Vec<u8>> {
        let storage_key = self.raw_key_to_storage_key(key_raw);
        if env::storage_write(&storage_key, value_raw) {
            Some(env::storage_get_evicted().unwrap())
        } else {
            None
        }
    }

    /// Removes a serialized key from the map, returning the serialized value at the key if the key
    /// was previously in the map.
    pub fn remove_raw(&mut self, key_raw: &[u8]) -> Option<Vec<u8>> {
        let storage_key = self.raw_key_to_storage_key(key_raw);
        if env::storage_remove(&storage_key) {
            Some(env::storage_get_evicted().unwrap())
        } else {
            None
        }
    }
```

LookupMap is not aware of its key-value pairs. They are stored directly on the contract memory which is also a map. The key for storing values is "LopkupMap prefix + provided key". It is not possible to clear the Lookup map without passing the keys.